File: backend/internal/store/meetups.go
package store

import "sync"

// Meetup represents a boardgame session
type Meetup struct {
	ID            string  `json:"id"`
	Title         string  `json:"title"`
	Game          string  `json:"game"`
	HostName      string  `json:"host_name"`
	HostUID       string  `json:"host_uid"`
	Lat           float64 `json:"lat"`
	Lng           float64 `json:"lng"`
	PlayersCount  int     `json:"players_count"`
	PlayersNeeded int     `json:"players_needed"`
	Time          string  `json:"time"`
	Color         string  `json:"color"`
}

// MeetupStore is a thread-safe in-memory database for meetups
type MeetupStore struct {
	sync.RWMutex
	meetups []Meetup
}

// NewMeetupStore initializes store with default values
func NewMeetupStore() *MeetupStore {
	return &MeetupStore{
		meetups: []Meetup{
			{
				ID:            "1",
				Title:         "Hội Ma Sói Đêm Trăng Q1",
				Game:          "Ultimate Werewolf",
				HostName:      "Minh Tuấn",
				HostUID:       "default-host-1",
				Lat:           10.7769,
				Lng:           106.7009,
				PlayersCount:  11,
				PlayersNeeded: 15,
				Time:          "2026-07-10T19:30",
				Color:         "#bca0f5", // purple
			},
			{
				ID:            "2",
				Title:         "Sân Chơi Mèo Nổ Q3",
				Game:          "Exploding Kittens",
				HostName:      "Thanh Trúc",
				HostUID:       "default-host-2",
				Lat:           10.7828,
				Lng:           106.6896,
				PlayersCount:  4,
				PlayersNeeded: 5,
				Time:          "2026-07-11T15:00",
				Color:         "#ffa4b2", // pink
			},
			{
				ID:            "3",
				Title:         "CLB Cờ Tỷ Phú Bình Thạnh",
				Game:          "Monopoly Deal",
				HostName:      "Khánh Huy",
				HostUID:       "default-host-3",
				Lat:           10.7981,
				Lng:           106.7051,
				PlayersCount:  3,
				PlayersNeeded: 6,
				Time:          "2026-07-12T18:00",
				Color:         "#ffe869", // yellow
			},
			{
				ID:            "4",
				Title:         "Chiến Thần Catan Hoàn Kiếm",
				Game:          "Settlers of Catan",
				HostName:      "Hoàng Lâm",
				HostUID:       "default-host-4",
				Lat:           21.0285,
				Lng:           105.8542,
				PlayersCount:  2,
				PlayersNeeded: 4,
				Time:          "2026-07-11T19:00",
				Color:         "#9ee3b2", // green
			},
			{
				ID:            "5",
				Title:         "Hội Avalon Tây Hồ",
				Game:          "Avalon",
				HostName:      "Thu Giang",
				HostUID:       "default-host-5",
				Lat:           21.0588,
				Lng:           105.8285,
				PlayersCount:  5,
				PlayersNeeded: 10,
				Time:          "2026-07-12T14:30",
				Color:         "#a4f0fd", // cyan
			},
		},
	}
}

// GetAll returns a copy of all meetups
func (s *MeetupStore) GetAll() []Meetup {
	s.RLock()
	defer s.RUnlock()
	
	// Create a copy to prevent external mutation
	result := make([]Meetup, len(s.meetups))
	copy(result, s.meetups)
	return result
}

// Add appends a new meetup to the store
func (s *MeetupStore) Add(meetup Meetup) {
	s.Lock()
	defer s.Unlock()
	s.meetups = append(s.meetups, meetup)
}

File: backend/src/store/meetups.ts
export interface Meetup {
  id: string;
  title: string;
  game: string;
  host_name: string;
  host_uid: string;
  lat: number;
  lng: number;
  players_count: number;
  players_needed: number;
  time: string;
  color: string;
}

export class MeetupStore {
  private meetups: Meetup[] = [
    {
      id: "1",
      title: "Hội Ma Sói Đêm Trăng Q1",
      game: "Ultimate Werewolf",
      host_name: "Minh Tuấn",
      host_uid: "default-host-1",
      lat: 10.7769,
      lng: 106.7009,
      players_count: 11,
      players_needed: 15,
      time: "2026-07-10T19:30",
      color: "#bca0f5",
    },
    {
      id: "2",
      title: "Sân Chơi Mèo Nổ Q3",
      game: "Exploding Kittens",
      host_name: "Thanh Trúc",
      host_uid: "default-host-2",
      lat: 10.7828,
      lng: 106.6896,
      players_count: 4,
      players_needed: 5,
      time: "2026-07-11T15:00",
      color: "#ffa4b2",
    },
    {
      id: "3",
      title: "CLB Cờ Tỷ Phú Bình Thạnh",
      game: "Monopoly Deal",
      host_name: "Khánh Huy",
      host_uid: "default-host-3",
      lat: 10.7981,
      lng: 106.7051,
      players_count: 3,
      players_needed: 6,
      time: "2026-07-12T18:00",
      color: "#ffe869",
    },
    {
      id: "4",
      title: "Chiến Thần Catan Hoàn Kiếm",
      game: "Settlers of Catan",
      host_name: "Hoàng Lâm",
      host_uid: "default-host-4",
      lat: 21.0285,
      lng: 105.8542,
      players_count: 2,
      players_needed: 4,
      time: "2026-07-11T19:00",
      color: "#9ee3b2",
    },
    {
      id: "5",
      title: "Hội Avalon Tây Hồ",
      game: "Avalon",
      host_name: "Thu Giang",
      host_uid: "default-host-5",
      lat: 21.0588,
      lng: 105.8285,
      players_count: 5,
      players_needed: 10,
      time: "2026-07-12T14:30",
      color: "#a4f0fd",
    },
  ];

  public getAll(): Meetup[] {
    return [...this.meetups];
  }

  public add(meetup: Meetup): void {
    this.meetups.push(meetup);
  }
}

export const meetupStore = new MeetupStore();
